src/common/database.c

```diff
@@ -4185,6 +4185,32 @@ dt_database_t *dt_database_init(const char *alternative, const gboolean load_dat
   else
     snprintf(dbfilename_data, sizeof(dbfilename_data), ":memory:");
 
+  // It may happen that we will not have write access to the database restored
+  // from a backup or snapshot. Running darktable with a database that cannot
+  // be written to may result in incorrect operation, the cause of which will
+  // be difficult to diagnose. Let's check if we can continue.
+  char *readonly_db_text = g_markup_printf_escaped(
```

Comment:
Since this text is only used when the access() calls fail, it makes more sense to generate it inside the inner "if".
Strictly speaking, you should also free the string afterwards, but we're exiting anyway in that case.